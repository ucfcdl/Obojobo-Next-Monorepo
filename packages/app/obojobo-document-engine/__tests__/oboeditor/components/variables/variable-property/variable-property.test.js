import React from 'react'
import { shallow, mount } from 'enzyme'

import VariableProperties from '../../../../../src/scripts/oboeditor/components/variables/variable-property/variable-property'

describe('Variable Properties', () => {
	test('VariableProperties node', () => {
		const variable = {
			name: 'static_var',
			type: 'static-value',
			value: '3'
		}

		const onDeleteVariable = jest.fn()
		const component = shallow(
			<VariableProperties variable={variable} onDeleteVariable={onDeleteVariable} />
		)
		expect(component.html()).toMatchSnapshot()
	})

	test('VariableProperties clicks "delete" will call "onDeleteVariable"', () => {
		global.confirm = () => true

		const variable = {
			name: 'static_var',
			type: 'static-value',
			value: '3'
		}

		const onDeleteVariable = jest.fn()
		const component = mount(
			<VariableProperties variable={variable} onDeleteVariable={onDeleteVariable} />
		)

		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(onDeleteVariable).toHaveBeenCalled()
		expect(component.html()).toMatchSnapshot()
	})

	test('VariableProperties does not call "onDeleteVariable" when user cancels', () => {
		global.confirm = () => false

		const variable = {
			name: 'static_var',
			type: 'static-value',
			value: '3'
		}

		const onDeleteVariable = jest.fn()
		const component = mount(
			<VariableProperties variable={variable} onDeleteVariable={onDeleteVariable} />
		)

		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(onDeleteVariable).not.toHaveBeenCalled()
		expect(component.html()).toMatchSnapshot()
	})

	test('VariableProperties calls onChange when input changes', () => {
		const variable = {
			name: 'static_var',
			type: 'static-value',
			value: '3'
		}

		const onChange = jest.fn()
		const component = mount(<VariableProperties variable={variable} onChange={onChange} />)

		component
			.find('input')
			.at(0)
			.simulate('change', { target: { value: '333' } })

		expect(onChange).toHaveBeenCalled()
		expect(component.html()).toMatchSnapshot()
	})
})
