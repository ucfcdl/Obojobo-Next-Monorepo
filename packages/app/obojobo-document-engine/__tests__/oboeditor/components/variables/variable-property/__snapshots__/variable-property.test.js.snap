// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variable Properties VariableProperty calls onChange when input changes 1`] = `"<div class=\\"variable-property\\"><label class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"variable-property--input-item\\" name=\\"name\\" value=\\"static_var\\"><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain variable's name rules\\">?</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></label><label class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"variable-property--select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></label><div class=\\"variable-values\\"><label class=\\"variable-values--group\\"><label>Value: </label><input class=\\"variable-property--input-item\\" type=\\"text\\" name=\\"value\\" value=\\"3\\"></label></div><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div>"`;

exports[`Variable Properties VariableProperty clicks "delete" will call "deleteVariable" 1`] = `"<div class=\\"variable-property\\"><label class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"variable-property--input-item\\" name=\\"name\\" value=\\"static_var\\"><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain variable's name rules\\">?</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></label><label class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"variable-property--select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></label><div class=\\"variable-values\\"><label class=\\"variable-values--group\\"><label>Value: </label><input class=\\"variable-property--input-item\\" type=\\"text\\" name=\\"value\\" value=\\"3\\"></label></div><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div>"`;

exports[`Variable Properties VariableProperty does not call "deleteVariable" when user cancels 1`] = `"<div class=\\"variable-property\\"><label class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"variable-property--input-item\\" name=\\"name\\" value=\\"static_var\\"><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain variable's name rules\\">?</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></label><label class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"variable-property--select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></label><div class=\\"variable-values\\"><label class=\\"variable-values--group\\"><label>Value: </label><input class=\\"variable-property--input-item\\" type=\\"text\\" name=\\"value\\" value=\\"3\\"></label></div><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div>"`;

exports[`Variable Properties VariableProperty node 1`] = `"<div class=\\"variable-property\\"><label class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"variable-property--input-item\\" name=\\"name\\" value=\\"static_var\\"/><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain variable&#x27;s name rules\\">?</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></label><label class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"variable-property--select-item\\" name=\\"type\\"><option selected=\\"\\" value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></label><div class=\\"variable-values\\"><label class=\\"variable-values--group\\"><label>Value: </label><input type=\\"text\\" class=\\"variable-property--input-item\\" name=\\"value\\" value=\\"3\\"/></label></div><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div>"`;

exports[`Variable Properties VariableProperty node with invalid variable 1`] = `"<div class=\\"variable-property\\"></div>"`;

exports[`Variable Properties VariableProperty node without default name or type 1`] = `"<div class=\\"variable-property\\"><label class=\\"group-item\\"><label><strong>Name:</strong></label><input class=\\"variable-property--input-item\\" name=\\"name\\" value=\\"\\"/><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain variable&#x27;s name rules\\">?</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--delete-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Delete</button></div></label><label class=\\"group-item\\"><label><strong>Type:</strong></label><select class=\\"variable-property--select-item\\" name=\\"type\\"><option value=\\"static-value\\">Static Number of Text Value</option><option value=\\"random-number\\">Random Number</option><option value=\\"static-list\\">Static List</option><option value=\\"random-list\\">Random List</option><option value=\\"random-sequence\\">Random Sequence</option><option value=\\"pick-one\\">Pick One</option><option value=\\"pick-list\\">Pick List</option></select></label><p><strong>Duplicate:</strong></p><div class=\\"obojobo-draft--components--button is-not-dangerous align-center variable-property--duplicate-button\\"><button class=\\"button\\" contenteditable=\\"false\\">Create another variable like this one</button></div></div>"`;
